Assets/Scripts/Models/Parameter.cs

```diff
@@ -98,14 +98,20 @@ public string Value
 
     public static Parameter ReadXml(XmlReader reader)
     {
+        Parameter paramGroup = new Parameter(reader.GetAttribute("name"));
         XmlReader subReader = reader.ReadSubtree();
-        Parameter paramGroup = new Parameter(subReader.GetAttribute("name"));
 
         // Advance to the first inner element. Two reads are needed to ensure we don't get stuck on containing Element, or an EndElement
         subReader.Read();
+
+        // In case the reader gets passed early, we descend to Params if it's not a Params or Param
+        if (subReader.Name != "Params" && subReader.Name != "Param")
```

Comment:
No, it wouldn't, I am checking for precisely what I want. It is bad practice to check for something that only generally fits what you want and hope that nothing changes later that will cause false positives.

Further this makes it explicitly clear what this line is looking for.